swaybar/tray/dbusmenu.c

```diff
@@ -0,0 +1,1381 @@
+		free(slot);
+	}
+}
+
+static void swaybar_dbusmenu_get_layout_root(struct swaybar_dbusmenu *menu) {
+	swaybar_dbusmenu_get_layout(menu, 0);
+}
+
+static int get_icon_theme_path_callback(sd_bus_message *msg, void *data,
+                                        sd_bus_error *error) {
+	struct swaybar_sni_slot *slot = data;
+	struct swaybar_sni *sni = slot->sni;
+	wl_list_remove(&slot->link);
+	free(slot);
+
+	int ret;
+	if (!sd_bus_message_is_method_error(msg, NULL)) {
+		ret = sd_bus_message_enter_container(msg, 'v', NULL);
+		if (ret >= 0) {
+			ret = sd_bus_message_read_strv(msg, &sni->menu_icon_theme_paths);
+		}
+	} else {
+		ret = -sd_bus_message_get_errno(msg);
+	}
+
+	if (ret < 0) {
+		sway_log(SWAY_ERROR, "%s%s failed to read IconThemePath: %s", sni->service,
+			sni->menu, strerror(-ret));
+	}
+	return ret;
+}
+
+static void swaybar_dbusmenu_setup(struct swaybar_dbusmenu *menu) {
+	struct swaybar_sni_slot *slot = calloc(1, sizeof(struct swaybar_sni_slot));
+	slot->sni = menu->sni;
+	int ret = sd_bus_call_method_async( menu->sni->tray->bus, &slot->slot, 
+			menu->sni->service, menu->sni->path, "org.freedesktop.DBus.Properties",
+			"Get", get_icon_theme_path_callback, slot, "ss", menu->sni->interface,
+			"IconThemePath");
+	if (ret >= 0) {
+		wl_list_insert(&menu->sni->slots, &slot->link);
+	} else {
+		sway_log(SWAY_ERROR, "%s%s failed to get IconThemePath: %s",
+			menu->sni->service, menu->sni->menu, strerror(-ret));
+		free(slot);
+	}
+
+	swaybar_dbusmenu_setup_signals(menu);
+	swaybar_dbusmenu_get_layout_root(menu);
+}
+
+static void handle_global(void *data, struct wl_registry *registry,
+                          uint32_t name, const char *interface,
+                          uint32_t version) {
+	struct swaybar_dbusmenu *menu = data;
+	if (strcmp(interface, xdg_wm_base_interface.name) == 0) {
+		menu->wm_base = wl_registry_bind(registry, name, &xdg_wm_base_interface, 1);
+	}
+}
+
+static void handle_global_remove(void *data, struct wl_registry *registry,
+                                 uint32_t name) {
+  // intentionally left blank
+}
+
+static const struct wl_registry_listener registry_listener = {
+    .global = handle_global,
+	.global_remove = handle_global_remove,
+};
+
+struct swaybar_dbusmenu *swaybar_dbusmenu_create(struct swaybar_sni *sni,
+                                                 struct swaybar_output *output,
+                                                 struct swaybar_seat *seat,
+                                                 uint32_t serial, int x,
+                                                 int y) {
+	struct swaybar_dbusmenu *dbusmenu = calloc(1, sizeof(struct swaybar_dbusmenu));
+
+	if (!dbusmenu) {
+		sway_log(SWAY_DEBUG, "Could not allocate dbusmenu");
+		return NULL;
+	}
+
+	sni->tray->menu = dbusmenu;
+
+	dbusmenu->sni = sni;
+	dbusmenu->output = output;
+	dbusmenu->seat = seat;
+	dbusmenu->serial = serial;
+	dbusmenu->x = seat->pointer.x;
+	dbusmenu->y = seat->pointer.y;
```

Comment:
I imagine this was supposed to use the `x` and `y` arguments? Otherwise, remove them.